frappe_telegram/frappe_telegram/doctype/telegram_bot/test_telegram_bot.py

```diff
@@ -1,8 +1,137 @@
 # Copyright (c) 2021, Leam Technology Systems and Contributors
 # See license.txt
 
-# import frappe
 import unittest
 
+import frappe
+from frappe_telegram.frappe_telegram.doctype.telegram_bot import DEFAULT_TELEGRAM_BOT_KEY
+from frappe_telegram.utils.test_fixture import TestFixture
+
+
+class TelegramBotFixtures(TestFixture):
+    def __init__(self):
+        super().__init__()
+        self.DEFAULT_DOCTYPE = "Customer Document Type"
+
+    def make_fixtures(self):
+
+        # Some fixtures that have not been inserted
+
+        fix1 = frappe.get_doc(frappe._dict(
+            doctype="Telegram Bot",
+            title="TestBotFix1",
+            api_token="randonapitoken"
+        ))
+        self.add_document(fix1)
+
+        fix2 = frappe.get_doc(frappe._dict(
+            doctype="Telegram Bot",
+            title="TestBotFix2",
+            api_token="otherrandonapitoken"
+        ))
+        self.add_document(fix2)
+
+        fix3 = frappe.get_doc(frappe._dict(
+            doctype="Telegram Bot",
+            title="TestBotFix3",
+            api_token="anotherrandonapitoken"
+        ))
+        self.add_document(fix3)
+
+
 class TestTelegramBot(unittest.TestCase):
-	pass
+
```

Comment:
I am getting an error while running this:
```bash
$ bench run-tests --doctype "Telegram Bot"
```
<details>
<summary>Output</summary>

```py
Traceback (most recent call last):
  File "/home/fahimalizain/.pyenv/versions/3.7.9/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/fahimalizain/.pyenv/versions/3.7.9/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/fahimalizain/dev-bench/frappe-telegram/apps/frappe/frappe/utils/bench_helper.py", line 104, in <module>
    main()
  File "/home/fahimalizain/dev-bench/frappe-telegram/apps/frappe/frappe/utils/bench_helper.py", line 19, in main
    click.Group(commands=commands)(prog_name='bench')
  File "/home/fahimalizain/dev-bench/frappe-telegram/env/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/fahimalizain/dev-bench/frappe-telegram/env/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/fahimalizain/dev-bench/frappe-telegram/env/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/fahimalizain/dev-bench/frappe-telegram/env/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/fahimalizain/dev-bench/frappe-telegram/env/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/fahimalizain/dev-bench/frappe-telegram/env/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/fahimalizain/dev-bench/frappe-telegram/env/lib/python3.7/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/fahimalizain/dev-bench/frappe-telegram/apps/frappe/frappe/commands/__init__.py", line 27, in _func
    ret = f(frappe._dict(ctx.obj), *args, **kwargs)
  File "/home/fahimalizain/dev-bench/frappe-telegram/apps/frappe/frappe/commands/utils.py", line 583, in run_tests
    ui_tests=ui_tests, doctype_list_path=doctype_list_path, failfast=failfast)
  File "/home/fahimalizain/dev-bench/frappe-telegram/apps/frappe/frappe/test_runner.py", line 70, in main
    ret = run_tests_for_doctype(doctype, verbose, tests, force, profile, failfast=failfast, junit_xml_output=junit_xml_output)
  File "/home/fahimalizain/dev-bench/frappe-telegram/apps/frappe/frappe/test_runner.py", line 174, in run_tests_for_doctype
    return _run_unittest(modules, verbose=verbose, tests=tests, profile=profile, failfast=failfast, junit_xml_output=junit_xml_output)
  File "/home/fahimalizain/dev-bench/frappe-telegram/apps/frappe/frappe/test_runner.py", line 186, in _run_unittest
    frappe.db.begin()
  File "/home/fahimalizain/dev-bench/frappe-telegram/apps/frappe/frappe/database/database.py", line 752, in begin
    self.sql("START TRANSACTION")
  File "/home/fahimalizain/dev-bench/frappe-telegram/apps/frappe/frappe/database/database.py", line 126, in sql
    self.check_transaction_status(query)
  File "/home/fahimalizain/dev-bench/frappe-telegram/apps/frappe/frappe/database/database.py", line 265, in check_transaction_status
    raise Exception('This statement can cause implicit commit')
Exception: This statement can cause implicit commit
```
</details>

It may not be related to this PR, can you please provide pointers on this ?

Reply:
I was able to replicate this using the `--doctype` option, but hmm I'm not sure. Might be an issue with frappe. The tests run properly with both the `--app` and `--module` options...